### DISCUSSION TITLE
Database in Discord bot

### DISCUSSION THREAD

**Discussion by sarim-developer** [community/community#136590] [Programming Help]
### Select Topic Area

Question

### Body

What database should i use instead of MongoDB atlas and postreSQL

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @sarim-devlopers ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by erencanbas** [comment]
**1. SQLite:**

- **Why Use It:** Lightweight and serverless; ideal for small to medium-scale bots. It’s a self-contained database that stores data in a single file, which is perfect for bots that don’t require complex queries or large datasets.
- **Pros**: Easy to set up, no need for a server, minimal configuration.
- **Cons**: Not suitable for very large-scale applications or highly concurrent environments.

**2. Redis:**

- **Why Use It**: Redis is an in-memory data store often used for caching but can be used as a database. It’s great for high-performance needs like leaderboards, counters, or temporary data.
- **Pros**: Extremely fast, supports key-value storage, has built-in pub/sub capabilities.
- **Cons**: Limited data persistence, better for specific use cases rather than general-purpose data storage.

**3. Firebase Realtime Database / Firestore:**

- **Why Use It**: Managed NoSQL databases by Google, designed for real-time applications. Firestore offers more structured and scalable data management.
- **Pros**: Easy to integrate, handles real-time updates well, scalable.
- **Cons**: Paid service with usage limits, might be overkill if your bot doesn’t require real-time features.

**4. MySQL / MariaDB:**

- **Why Use It:** Reliable, mature relational databases. If you prefer SQL but want something different from PostgreSQL, MySQL or its open-source fork MariaDB are good alternatives.
- **Pros**: Well-documented, widely supported, suitable for large-scale applications.
- **Cons**: Requires server setup, can be overkill for small bots.

**5. LevelDB:**

- **Why Use It:** Embedded key-value store library. Good for smaller applications or when you need a simple storage solution.
- **Pros**: Lightweight, simple to use, no need for a server.
- **Cons**: Limited querying capabilities, not suited for complex data relationships.

**6. DynamoDB (AWS):**

- **Why Use It:** A managed NoSQL database by AWS, suitable for scaling up. Works well for distributed systems.
- **Pros**: Managed by AWS, scalable, integrated well with other AWS services.
- **Cons**: Requires familiarity with AWS, can become costly at scale.
